lib/babe/babe.go

```diff
@@ -497,6 +497,21 @@ func (b *Service) handleSlot(epoch, slotNum uint64,
 		return err
 	}
 
+	slotNumOfTheBestBlock, err := b.blockState.GetSlotForBlock(parentHeader.Hash())
+	if err != nil {
+		return fmt.Errorf("could not get slot for block %s: %w", parentHeader.Hash(), err)
```

Comment:
nit

- do not prefix error wrappings with `failed`/`cannot`/`could not` since it builds up repetition for deeply wrapped errors
- the parent header hash context should be wrapped by `GetSlotForBlock`, not here

```suggestion
		return fmt.Errorf("getting slot for block: %w", err)
```

Reply:
> since it builds up a repetition for deeply wrapped errors

I don't think so, would be much easier to read eg. `cannot build block: cannot get slot number: failed to get block from the tree` than `cannot build block: getting slot number: retrieving block from the tree`, I think writing better log messages is needed but in this case I think we should keep as it is